### PR TITLE
Move non-sensitive values from secrets to inputs

### PR DESCRIPTION
**Pull request by stuartleeks**
Improve log output for PR builds by moving non-sensitive values to inputs
(Note: any value still derived from repo secrets will remain masked e.g. for builds on main)

part of #1538

### CODE REVIEW

**Comment by stuartleeks** [comment]
Previous successful run via `deploy_tre_branch.yml`

File: .github/workflows/deploy_tre.yml [hunk 1]
```diff
@@ -22,13 +22,15 @@ jobs:
     uses: ./.github/workflows/deploy_tre_reusable.yml
     with:
       ciGitRef: ${{ github.ref }}
+      ACR_NAME: ${{ secrets.ACR_NAME }}
```
Comment (ross-p-smith) [inline .github/workflows/deploy_tre.yml:25]: This is a globally unique acr - do we really want that displayed in our logs? I toyed with this in the past, but I think it's better that the public dont see it.

Comment (ross-p-smith) [reply .github/workflows/deploy_tre.yml:25]: So if it's still masked, what benefit does this change have?

Comment (stuartleeks) [reply .github/workflows/deploy_tre.yml:25]: For builds on `main`, none. But the `deploy_tre_reusable.yml` is also used from the PR builds, so making it an input there reduces masking on PR builds (where `ACR_NAME` points to the specific instance for the PR deployment)

**Review by ross-p-smith — APPROVED** [review]

**Comment by stuartleeks** [comment]
Redeployed using `deploy_tre_branch.yml` after resolving merge conflicts: https://github.com/microsoft/AzureTRE/runs/5624442302?check_suite_focus=true

**Pull request merged**